src/scripts/activate-user.ts

```diff
@@ -0,0 +1,197 @@
+ */
+const validateUserStatusPriorToActivation = (
+  user: User
+): TE.TaskEither<string, ActivatableUser> => {
+  const userStatus = user.status;
+  const activeStatus = okta.UserStatus.ACTIVE;
+  const context = `User [${user.id}] [${user.email}] has status [${userStatus}]. Activation is reserved for users with status ${okta.UserStatus.STAGED} or ${okta.UserStatus.DEPROVISIONED}.`;
+
+  // eslint-disable-next-line sonarjs/no-small-switch, functional/no-conditional-statement
+  switch (userStatus) {
+    case okta.UserStatus.ACTIVE: {
+      return TE.left(context);
+    }
+    case okta.UserStatus.PROVISIONED: {
+      return TE.left(
+        `${context} To transition user to ${activeStatus} status, please follow through with the activation workflow.`
+      );
+    }
+
+    case okta.UserStatus.LOCKED_OUT: {
+      return TE.left(
+        `${context} To transition user to ${activeStatus} status, please use the unlock command.`
+      );
+    }
+    case okta.UserStatus.PASSWORD_EXPIRED: {
+      return TE.left(
+        `${context} To transition user to ${activeStatus} status, please instruct user to login with temporary password and follow the password reset process.`
+      );
+    }
+    case okta.UserStatus.RECOVERY: {
+      return TE.left(
+        `${context} To transition user to ${activeStatus} status, please follow through with the activation workflow or restart the workflow using the reactivate-user command.`
+      );
+    }
+    case okta.UserStatus.SUSPENDED: {
+      return TE.left(
+        `${context} To transition user to ${activeStatus} status, please use the unsuspend-user command.`
+      );
+    }
+    // STAGED or DEPROVISIONED user status
+    default: {
+      return TE.right({
+        ...user,
+        status: userStatus,
+      });
+    }
+  }
+};
+
+/**
+ * Prints out what would happen if we were to activate the user.
+ * @param user - the user to dry run the activation for.
+ * @returns a TaskEither that resolves to the user.
+ */
+const dryRunActivateUser = (
+  user: ActivatableUser
+): TE.TaskEither<string, User> => {
+  // eslint-disable-next-line functional/no-expression-statement
+  console.info(
+    `User current status is [${user.status}]. Will attempt to activate [${user.id}] [${user.email}].`
+  );
+  return TE.right(user);
+};
+
+/**
+ * Activates the user.
+ * @param service - the service to use to activate the user.
+ * @param user - the user to activate.
+ * @returns a TaskEither that resolves to the activated user.
+ */
+const activateUser = (
+  service: UserService,
+  user: ActivatableUser
+): TE.TaskEither<string, User> => {
+  // eslint-disable-next-line functional/no-expression-statement
+  console.info(`User current status is [${user.status}].`);
+  return pipe(
+    service.activateUser(user.id),
+    TE.tapIO((user) => Console.info(`Activated [${user.id}] [${user.email}].`))
```

Comment:
```suggestion
    TE.tapIO((user) => Console.info(`Activated [${user.id}] [${user.email}], new status is [???].`))
```

Reply:
modified in commit: e743c8258d8660e8602d291c16dd3af5b11fbc97

Reply:
This won't show the updated status. A separate getUser is needed to see the newly updated status so i did this:
>const activateUser = (
  service: UserService,
  user: ActivatableUser
): TE.TaskEither<string, User> =>
  pipe(
    Console.info(
      `Activating user [${user.id}] [${user.email}] with status [${user.status}]...`
    ),
    TE.rightIO,
    // eslint-disable-next-line functional/functional-parameters
    TE.chain(() => service.activateUser(user.id)),
    TE.chain((user) => validateUserExist(service, user.id)),
    TE.tapIO((user) =>
      Console.info(
        `Activated [${user.id}] [${user.email}], new status is [${user.status}].`
      )
    )
  );

Reply:
Nice.

In the future, let's consider whether the service method (e.g. `activateUser`) should be responsible for this.

This is good for now.

Reply:
I added logs for fetching user to inform the user of the 2 fetches. It looks like this now: 
![image](https://github.com/agiledigital-labs/oktagon/assets/124639768/0c61a605-e0d4-4c1c-8815-37b193cccc62)